src/framework/core/js/Fluid.js

```diff
@@ -205,27 +203,26 @@ var fluid = fluid || fluid_3_0_0;
      * Signals an error to the framework. The default behaviour is to log a structured error message and throw an exception. This strategy may be configured using the legacy
      * API <code>fluid.pushSoftFailure</code> or else by adding and removing suitably namespaced listeners to the special event <code>fluid.failureEvent</code>
      *
-     * @param {String} message - The error message to log.
+     * @param {...String} messages - The error messages to log.
      *
      * All arguments after the first are passed on to (and should be suitable to pass on to) the native console.log
      * function.
      */
-    fluid.fail = function (/* message, ... */) {
-        var args = fluid.makeArray(arguments);
+    fluid.fail = function (...messages) {
         var activity = fluid.makeArray(fluid.describeActivity()); // Take copy since we will destructively modify
         fluid.popActivity(activity.length); // clear any current activity - TODO: the framework currently has no exception handlers, although it will in time
         if (fluid.failureEvent) { // notify any framework failure prior to successfully setting up the failure event below
-            fluid.failureEvent.fire(args, activity);
+            fluid.failureEvent.fire(messages, activity);
         } else {
-            fluid.logFailure(args, activity);
-            fluid.builtinFail(args, activity);
+            fluid.logFailure(messages, activity);
+            fluid.builtinFail(messages, activity);
```

Comment:
@amb26 I want to highlight this change to make sure you see it. There is another similar one elsewhere in the code as well. The change here is to use the [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters) in stead of arguments. The rest parameter returns a named array of the arguments from that position.

Reply:
I assume that the use of the spread pattern makes the use of fluid.makeArray() unnecessary on the first line

Reply:
@amb26 yes, the call to `fluid.makeArray` that had been on line 214 was removed for this reason. The call to `fluid.makeArray` on line 212 (formerly 215) seems be used for making a copy.